### ISSUE TITLE
build failed for OSX in release mode

### ISSUE THREAD

**Issue by SiminBadri** [WolfEngine/WolfEngine#274]
Building in release mode fails in OSX. The library `mimalloc` is built successfully in release mode, however, linking to it causes failure in the compilation of wolf.

**Comment by PooyaEimandar** [comment]
> Building in release mode fails in OSX. The library `mimalloc` is built successfully in release mode, however, linking to it causes failure in the compilation of wolf.

kindly make sure pull the latest commit.

**Comment by SiminBadri** [comment]
When I run the command `cargo run --release`, the building fails with the following error:

```
warning: /Users/siminbadri/ALPHA/Wolves/WolfEngine/wolf/Cargo.toml: unused manifest key: build
   Compiling wolfengine v3.0.3 (/Users/siminbadri/ALPHA/Wolves/WolfEngine/wolf)
error: could not find native static library `mimalloc-minsizerel`, perhaps an -L flag is missing?

error: could not compile `wolfengine` due to previous error
```

Running the above command changes the `OPT_LEVEL` environment variable to either `"s"` or `"z"`. This change forces the `cmake` to build the library for `MinSizeRel` mode. Regarding the file `wolf/build.rs`, the library that will be linked to wolf should have the name `mimalloc-minsizerel`. However, the name of the built library of `mimalloc` is `libmimalloc.a` 

Note that this built library is located at the path: 
`/WolfEngine/wolf/sys/build/MinSizeRel/_deps/mimalloc-static-build/libmimalloc.a`

**Comment by SiminBadri** [comment]
> > Building in release mode fails in OSX. The library `mimalloc` is built successfully in release mode, however, linking to it causes failure in the compilation of wolf.
> 
> kindly make sure pull the latest commit.

I cloned this repo with depth set to 1 and then tried to build it. In debug mode, the wolf is built successfully and the demo runs too. However, building in release mode fails with the error mentioned [here](https://github.com/WolfEngine/WolfEngine/issues/274#issuecomment-1124179743).

**Comment by SiminBadri** [comment]
It seems that linking to `mimalloc` is useless. I changed the file `wolf/build.rs` and removed the linking to the library `mimalloc`. Now, `wolf` is built and running successfully in both release and debug mode for osx. 

I applied this change as a pull request in the commit [here](https://github.com/WolfEngine/WolfEngine/pull/275#issue-1233092148).

**Comment by SiminBadri** [comment]
This issue is fixed in this [commit](fd9f4c82f761cc7814a69df92d27a9186bd44d98).
